src/services/fileServices/YmlFileServices/ConfigService.ts

```diff
@@ -0,0 +1,75 @@
+import { err, ok, Result } from "neverthrow"
+
+import ConfigParseError from "@root/errors/ConfigParseError"
+import {
+  ConfigFileName,
+  HtmlConfigFileName,
+  YmlConfigFileName,
+} from "@root/types/configYml"
+import { PathInfo } from "@root/types/pages"
+import { Brand } from "@root/types/util"
+
+// eslint-disable-next-line import/prefer-default-export
+export class ConfigService {
+  isConfigFile = (
+    pathInfo: PathInfo
+  ): Result<ConfigFileName, ConfigParseError> =>
+    // NOTE: A file can be a config file in a few scenarios:
+    // 1. The file's extension is `.yml` (given by jekyll)
+    // 2. The file's extension is `.html` + it's rooted in a custom folder
+    // The second condition could be made more specific (resource room)
+    // but we'd then have an additional dep + require a github api call
+    // so we will rely on a less specific condition for now.
+    {
+      if (pathInfo.name.endsWith(".yml"))
+        return ok(this.extractYmlConfigFileName(pathInfo))
+
+      return this.isHtmlConfigFile(pathInfo).map((htmlPathInfo) =>
+        this.extractHtmlConfigFileName(htmlPathInfo)
+      )
+    }
+
+  private extractYmlConfigFileName = ({
+    name: rawName,
+    path,
+  }: PathInfo): YmlConfigFileName => {
+    const name = Brand.fromString<"YmlConfig">(rawName)
+    return {
+      name,
+      path,
+      kind: "YmlConfig",
```

Comment:
use enums

Reply:
i like unions more tbh, i've ran into type errors w/ enums in the past due to how they are implemented in TS and hence, i've avoided them like the plague ever since.

see [here](https://fettblog.eu/tidy-typescript-avoid-enums/) or [here](https://www.niraj.life/blog/avoid-using-enums-in-typescript) for more details  but if you feel strongly about this, i can change it

Reply:
@seaerchin That's a good point! I think I used to do it like this:

```
enum ESomeEnum {
    ValOne: "1",
    ValTwo: "2"
}

type EFinalType = ESomeEnum.ValOne | ESomeEnum.ValTwo
```

Then we use the EFinalType for the actual assertion

Reply:
this actually results in the same thing HAHA

Reply:
oh ps, i haven't had time to look at this due to being caught up in the release but this has the same effect because the discriminated union still only allows the same set of values.

Reply:
yes true, but felt it is cleaner with the use of discriminated unions + enums rather than strings